packages/sanity-plugin-nrkno-iframe-preview/README.md

```diff
@@ -0,0 +1,166 @@
+# @nrk/sanity-plugin-nrkno-iframe-preview
+
+Liveupdated iframe preview component for Sanity Studio, compatible with any framework.
+
+Queries are executed by Sanity Studio and passed to the iframe. 
+This has the beneficial in that the render-app in the iframe
+does not need to listen for sanity client updates itself.
+
+![preview.png](./docs/images/preview.png)
+_Figure 1: IFramePreview with desktopMinWidth configured_
+
+The app running in the iframe should use `@nrk/nrkno-iframe-preview-api`, 
+which allows it to specify a GROQ-query for the Studio to execute.
+
+The result of the query will be sent to the iframe whenever the query
+revision matches the studio revision (ie. whenever the data changes).
+
+This means that the render app does not need to listen for changes itself, as
+data flows directly from the Studio via iframe.postMessage.
+
+# Installation
+In Sanity studio project run:
+
+`npx sanity install @nrk/sanity-plugin-nrkno-iframe-preview`
+
+This will install & add the plugin to sanity.json plugin array.
+
+## At-a-glance
+
+* Use IFramePreview component in studio structure view.
+    * Configure with render-app preview-url.
+    * Optionally configure desktopMinWidth.
+* Use [nrkno-sanity-iframe-preview-api](../nrkno-iframe-preview-api/README.md) in the render-app.
+    * Configure a groq-query (or just use document directly from Sanity Studio as is).
+* Enjoy live-updated preview in Studio, with queries executed by the Studio on behalf of the render app.
+
+## Usage
+Should be used as part of Sanity Studio StructureBuilder code:
+```ts
+import { IFramePreview } from '@nrk/sanity-plugin-nrkno-iframe-preview'
+
+S.view
+  .component(IFramePreview)
+  .options({
+      url: (doc: SanityDocument) => 'iframe-url', // (doc) => (string | Promise<string>)
+      mapDocument: (doc: SanityDocument) => ({ _id: doc._id }),
+      desktopMinWidth: 900,
+   })   
+  .icon(EyeIcon)
+  .id("preview")
+  .title("Preview")
+```
+
+Read [IFramePreviewBasicProps](src/components/basic/IFramePreviewBasic.tsx) jsdocs for config details.
+
+### Customize display texts
+
+Create a wrapper component, and provide `DisplayTextsContext`:
+
+```tsx
+import React from 'react';
+import {
+  defaultDisplayTexts,
+  DisplayTexts,
+  DisplayTextsContext,
+  IFramePreview,
+  IFramePreviewProps,
+} from '@nrk/sanity-plugin-nrkno-iframe-preview';
+
+const texts: DisplayTexts = {
+  ...defaultDisplayTexts,
+  iframeTitle: 'Changed preview title',
+};
+
+export function TranslatedIFramePreview(props: IFramePreviewProps) {
+  return (
+    <DisplayTextsContext.Provider value={texts}>
+      <IFramePreview {...props} />
+    </DisplayTextsContext.Provider>
+  );
+}
+```
+
+### Define preview component from schema
```

Comment:
Hva er fordelen med å gjøre dette fremfor slik som i det aller første eksempelet (`S.view.component(IFramePreview)...`)?

Ettersom koden i structure.ts ble ganske kompleks, syntes jeg det gir mer mening å bare ha mer logikk i `url: (doc: SanityDocument) => '<riktig iframeurl for SanityDocument>'`.

Om ikke det er noen store fordeler og dette er en "sær greie" ville jeg like en liten advarsel på seksjonen om at dette er for spesielle use-caser

Reply:
Mind the context her. Dette er i repoet der alt handler om option driven develoment. Jeg kan lenke til toppdocsene for dette. Men hele poenget med alt i  repoet er å komme seg bort fra "litt greier i mange filer som angår samme skjema".

Med views må du konfigurere et view per skjema og det skalerer veldig dårlig.
Skal se om jeg kan gi det litt mer mening med en intro to why.

Når det er sagt så kommer jeg nok til å slette hele avsnittet når jeg komemr med nrkno-sanity-structure lib, for da er alt dette håndtert (kommer i morgen eller onsdag).